File: 3034-points-that-intersect-with-cars/points-that-intersect-with-cars.cpp
class Solution {
public:
    int numberOfPoints(vector<vector<int>>& nums) {

        int n = nums.size();

        sort(begin(nums), end(nums));
        int mini = nums[0][0], maxi = nums[0][1];
        int ans = 0;
        for (int i = 0; i < n; i++) {
            if (nums[i][0] <= maxi) {
                // Overlap
                

                maxi=max(maxi,nums[i][1]);
            } else {
                ans += (maxi - mini + 1);
                mini = nums[i][0];
                maxi = nums[i][1];
            }
        }
        ans += (maxi - mini + 1);
        return ans;
    }
};

File: 3034-points-that-intersect-with-cars/points-that-intersect-with-cars.java

class Solution {
    public int numberOfPoints(List<List<Integer>> nums) {
        int n = nums.size();

        // Sort based on starting point
       nums.sort(Comparator.comparingInt(a -> a.get(0)));
    
        int mini = nums.get(0).get(0);
        int maxi = nums.get(0).get(1);
        int ans = 0;

        for (int i = 0; i < n; i++) {
            int start = nums.get(i).get(0);
            int end = nums.get(i).get(1);

            if (start <= maxi) {
                // Overlap
                maxi = Math.max(maxi, end);
            } else {
                ans += (maxi - mini + 1);
                mini = start;
                maxi = end;
            }
        }

        ans += (maxi - mini + 1);
        return ans;
    }
}
